src/modules/billing-plans/billing-plan.controller.ts

```diff
@@ -1,42 +1,71 @@
-import { Controller, Get, Param, Post, Body, UseGuards } from '@nestjs/common';
-import { ApiBearerAuth, ApiOperation, ApiResponse, ApiTags, ApiBody } from '@nestjs/swagger';
+import {
+  Controller,
+  Get,
+  Param,
+  Post,
+  Body,
+  UseGuards,
+  Patch,
+  ParseUUIDPipe,
+  Delete,
+  HttpCode,
+  HttpStatus,
+} from '@nestjs/common';
+import { ApiTags } from '@nestjs/swagger';
 import { SuperAdminGuard } from '../../guards/super-admin.guard';
 import { BillingPlanService } from './billing-plan.service';
 import { skipAuth } from '../../helpers/skipAuth';
 import { BillingPlanDto } from './dto/billing-plan.dto';
+import {
+  createBillingPlanDocs,
+  deleteBillingPlan,
+  getAllBillingPlansDocs,
+  getSingleBillingPlan,
+  updateBillingPlan,
+} from './docs/billing-plan-docs';
+import { UpdateBillingPlanDto } from './dto/update-billing-plan.dto';
 
 @ApiTags('Billing Plans')
 @Controller('billing-plans')
 export class BillingPlanController {
   constructor(private readonly billingPlanService: BillingPlanService) {}
 
   @Post('/')
+  @createBillingPlanDocs()
   @UseGuards(SuperAdminGuard)
-  @ApiBearerAuth()
-  @ApiOperation({ summary: 'Create billing plans' })
-  @ApiBody({ type: BillingPlanDto })
-  @ApiResponse({ status: 201, description: 'Billing plan created successfully.', type: BillingPlanDto })
-  @ApiResponse({ status: 200, description: 'Billing plan already exists in the database.', type: [BillingPlanDto] })
   async createBillingPlan(@Body() createBillingPlanDto: BillingPlanDto) {
     return this.billingPlanService.createBillingPlan(createBillingPlanDto);
   }
 
   @skipAuth()
+  @getAllBillingPlansDocs()
   @Get('/')
-  @ApiOperation({ summary: 'Get all billing plans' })
-  @ApiResponse({ status: 200, description: 'Billing plans retrieved successfully.', type: [BillingPlanDto] })
-  @ApiResponse({ status: 404, description: 'No billing plans found.' })
   async getAllBillingPlans() {
     return this.billingPlanService.getAllBillingPlans();
   }
 
   @skipAuth()
+  @getSingleBillingPlan()
   @Get('/:id')
-  @ApiOperation({ summary: 'Get single billing plan by ID' })
-  @ApiResponse({ status: 200, description: 'Billing plan retrieved successfully', type: BillingPlanDto })
-  @ApiResponse({ status: 400, description: 'Invalid billing plan ID' })
-  @ApiResponse({ status: 404, description: 'Billing plan not found' })
   async getSingleBillingPlan(@Param('id') id: string) {
     return this.billingPlanService.getSingleBillingPlan(id);
   }
+
+  @UseGuards(SuperAdminGuard)
+  @updateBillingPlan()
```

Comment:
Name this as uodateBillingPlanDoc so it's more descriptive

Reply:
done